### PR TITLE
Worked on graph and map info modules

### PR DESCRIPTION
**Pull request by perryjoy**

### CODE REVIEW

File: code/core/viewer.cpp [hunk 1]
```diff
@@ -85,30 +83,6 @@ void viewer::AddSelectable(QString id)
     mapScene->addItem(selectableItems[id]);
 }
 
-void viewer::HighlightShop(QString id)
```
Comment (kirillova-ak5) [inline code/core/viewer.cpp]: why are you delete this?

File: code/core/viewer.cpp [hunk 2]
```diff
@@ -244,49 +275,28 @@ void viewer::ViewGraph()
...
-    std::vector<vertex_graph> path = w.SearchWay(start, finish);
-    mapPic->setZValue(0);
-    SetAntialiasing(1);
-
-    for (int i = 0; i < path.size() - 1; ++i)
-    {
-        mapScene->addLine(path[i].vertexCoordinates.x, path[i].vertexCoordinates.y,
-                          path[i + 1].vertexCoordinates.x, path[i + 1].vertexCoordinates.y, color);
-    }
+//    graph w;
```
Comment (kirillova-ak5) [inline code/core/viewer.cpp]: isnt good

File: code/core/viewer.cpp [hunk 3]
```diff
@@ -330,21 +340,11 @@ bool viewer::viewportEvent(QEvent *event)
             //For Arina
             //On single touch event.
             //If it looks strange, you can move this block from "Touch Begin" to "TouchUpdate" or "TouchEnd"
-
```
Comment (kirillova-ak5) [inline code/core/viewer.cpp:333]: why are you delete this

File: code/device/fine_graph.h [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+{
+    point pos;
+    bool isShowable;
+    QString svgId;
+    std::vector<quint32> edgesConnected;
+    std::vector<mulifloor_connection> adjacentVertexesOnOtherFloors;
+
+    vertex();
+
+    double pathLen;
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:61]: are u sure vertex struct need to know something about path?

File: code/device/fine_graph.h [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+
+
+
+
+struct mall_floor
+{
+    std::vector<edge> edges; // edges on this floor
+    std::map<point, vertex> vertexes;
+
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:73]: very big space

File: code/device/fine_graph.h [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+   // PATHING
+
+    std::set<point> pointsToMap;
+
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:78]: very very big im scared

File: code/device/fine_graph.h [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+};
+
+
+
+
+class fine_graph : public QObject
+{
+    Q_OBJECT
+public:
+    //graph_alternative();
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:99]: commented code

File: code/device/fine_graph.h [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    point start;
+    quint32 startFloorIndex;
+
+    point end;
+    quint32 endFloorIndex;
+
+    path currentPath;
+
+    // inner functions
+
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:127]: very very big im scared

File: code/device/fine_graph.h [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    friend fine_graph CreateGraphFromSvg (QString picFileName);
+    friend class graph_parser;
+
+    // those are really bad, althoug it guaratees that you cant change parser without checking in here
+    friend void LoadGraphObjectsOfTheFloor (fine_graph* res, QDomNode& pathGroupHolder);
+    friend void LoadEdge (fine_graph* g, quint32 floor, QDomElement& e);
+    friend void LoadElevator (fine_graph* g, quint32 floor, QDomElement& e);
+    friend void LoadLadder (fine_graph* g, quint32 floor, QDomElement& l);
+
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:142]: why

File: code/device/fine_graph.h [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+};
+
+void RoundCoord (point& p);
+
+#ifndef GRAPH_H
```
Comment (kirillova-ak5) [inline code/device/fine_graph.h:148]: what

File: code/device/graph.h [hunk 11]
```diff
@@ -1,8 +1,10 @@
 #ifndef GRAPH_H
 #define GRAPH_H
 
+
```
Comment (kirillova-ak5) [inline code/device/graph.h:4]: empty line added

File: code/device/graph.h [hunk 12]
```diff
@@ -1,8 +1,10 @@
 #ifndef GRAPH_H
 #define GRAPH_H
 
+
 #include <vector>
-#include <string>
+
```
Comment (kirillova-ak5) [inline code/device/graph.h:6]: 2 emptylines added

File: code/device/mainwindow.h [hunk 13]
```diff
@@ -80,4 +80,5 @@ private slots:
     manager &manager;
 };
 
+
```
Comment (kirillova-ak5) [inline code/device/mainwindow.h:83]: why

File: code/device/viewer.h [hunk 14]
```diff
@@ -6,9 +6,9 @@
 #include "map.h"
 
 class coord;
+class path;
```
Comment (kirillova-ak5) [inline code/device/viewer.h]: is this struct?

File: code/device/viewer.h [hunk 15]
```diff
@@ -36,21 +36,20 @@ class viewer: public QGraphicsView
...
     void ChangeBgrLayer(QString id);
-
     void Clear();
     void AddLabel(QString text, int x, int y, QString idToLabeling = "", QWidget *parent = nullptr);
+    void AddPolyline(path* currentPath, quint32 currentFloorIndex);
+    void ClearPolyline();
     void ClearLabels();
+    void ClearUnstableVisible();
     void ClearSelectables();
-    void ClearHighlited();
```
Comment (kirillova-ak5) [inline code/device/viewer.h:53]: why are you delete this

File: code/device/viewer.h [hunk 16]
```diff
@@ -75,13 +74,10 @@ class viewer: public QGraphicsView
 
     std::map<QString, QGraphicsSvgItem*> unstableVisibleItems;
     std::map<QString, QGraphicsSvgItem*> selectableItems;
-
-    QGraphicsSvgItem* recoloredItem;    // Later I will make it to vector. IDD
```
Comment (kirillova-ak5) [inline code/device/viewer.h:79]: why are you delete this

File: code/device/fine_graph.cpp [hunk 17]
```diff
@@ -0,0 +1,812 @@
...
+    ClearPath();
+}
+
+vertex::vertex(): pos(), isShowable(false), svgId (""), edgesConnected(),adjacentVertexesOnOtherFloors (), pathLen(MAX_DBL), pathThroughFloorsLen (MAX_DBL)
+{
+
+}
+
+
+//bool edge::CrossesOnStart(const edge &neigbour) const
```
Comment (kirillova-ak5) [inline code/device/fine_graph.cpp:376]: u can delete all of this

File: code/core/viewer.cpp (hunk 1 above)
Comment (perryjoy) [reply code/core/viewer.cpp]: Nevermind, lets keep it. IDK

File: code/core/viewer.cpp (hunk 2 above)
Comment (perryjoy) [reply code/core/viewer.cpp]: Lets just kill this method

**Pull request merged**